src/controller/CHIPDeviceController.cpp

```diff
@@ -1888,23 +1883,18 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
             ChipLogProgress(Controller, "Device does not support configurable regulatory location");
             regulatoryConfig = capability;
         }
-        static constexpr size_t kMaxCountryCodeSize = 3;
-        char countryCodeStr[kMaxCountryCodeSize]    = "XX";
-        size_t actualCountryCodeSize                = 2;
-
-#if CONFIG_DEVICE_LAYER
-        CHIP_ERROR status =
-            DeviceLayer::ConfigurationMgr().GetCountryCode(countryCodeStr, kMaxCountryCodeSize, actualCountryCodeSize);
-#else
-        CHIP_ERROR status = CHIP_ERROR_NOT_IMPLEMENTED;
-#endif
-        if (status != CHIP_NO_ERROR)
+
+        CharSpan countryCode;
+        const auto & providedCountryCode = params.GetCountryCode();
+        if (providedCountryCode.HasValue())
+        {
+            countryCode = providedCountryCode.Value();
+        }
+        else
```

Comment:
This may affect folks that are using the config manager to set the country code. Can this default first the to value set in the config manager before defaulting to XX?

Reply:
The problem is that I don't actually want to have commissioner depending on the configuration manager at all, since it may not even exist in a sane way on the commissioner...  Configuration manager for the most part, if not entirely, does not make sense for a pure client app.

Reply:
The config manager should never be required by Controller. It's only there for "devices" and comes from "device layer". It's a bit of a semantic gap. Really, the example SDK commissioner can set a default country code of XX.

Reply:
In practice, CONFIG_DEVICE_LAYER is always defined, so that doesn't really buy us anything.

I understand the concern about compatibility, I feel like we're still early enough in, and that this was hard enough to discover to start with, that removing the wart should be OK...

But if we feel that we need to keep it (and at that point that would mean keeping it forever, right?) I can deal with it...

Reply:
Sure, I suppose that's fine.